pkg/yang/entry.go

```diff
@@ -1145,11 +1169,18 @@ func (e *Entry) ApplyDeviate() []error {
 						deviatedNode.Config = TSUnset
 					}
 
-					if devSpec.Default != "" {
-						if devSpec.Default == deviatedNode.Default {
-							deviatedNode.Default = ""
-						} else {
-							appendErr(fmt.Errorf("%s: tried to deviate delete a default statement that doesn't exist or with a non-matching keyword", Source(e.Node)))
+					if len(devSpec.Default) > 0 {
+						switch {
+						case deviatedNode.IsLeafList():
+							// It is unclear from RFC7950 on how deviate delete works
+							// when there are duplicate leaf-list values.
+							appendErr(fmt.Errorf("%s: deviate delete on default statements unsupported for leaf-lists, please use replace instead", Source(e.Node)))
```

Comment:
I'm not sure that I understand the error discussed above. Entries in a `leaf-list` must be unique ([ref](https://datatracker.ietf.org/doc/html/rfc7950#section-7.7)), and `delete` must match a specific keyword, so if you have:

```yang

leaf-list foo {
  type string;
  default "one";
  default "two";
}
```

then:

```yang

deviation "/foo" {
   deviate delete {
     default "one";
   }
}
```

then the default of `foo` would become `["two"]` rather than `["one", "two"]` in the before case. If there is no matching `default` for one that is removed then this is an error.

Reply:
https://datatracker.ietf.org/doc/html/rfc7950#section-7.7
>    In configuration data, the values in a leaf-list MUST be unique.

What if it's state data? Then which one do we delete?

Unfortunately I couldn't find it in https://datatracker.ietf.org/doc/html/rfc7950#section-7.7.2, and when looking at pyang's behaviour, I hit this: https://github.com/mbj4668/pyang/issues/756.

Reply:
I agree that the behaviour can be implemented for config-false leaf-lists, so I've added a TODO to work on it after this PR. This PR is in someone else's repository so I prefer to open a PR for that after this is merged. We need to take care of erroring out when there are duplicate defaults.

Reply:
That sounds fine with me.

This whole idea that `config true` and `config false` leaf-lists have different approaches seems a problem to me -- it just creates more complexity in the language. I'd be OK with an interpretation that just enforces uniqueness for config true and config false leaflists the same way.